examples/market_making_strategies/strategies/naive_market_making/naive_market_making.py

```diff
@@ -0,0 +1,188 @@
+      for order in open_orders.orders:
+        if order.order_side == "buy":
+          self.buy_for.add(order)
+        else:
+          self.buy_against.add(order)
+
+      new_buy_for_prices = []
+      new_buy_against_prices = []
+      new_buy_for_quantities = []
+      new_buy_against_quantities = []
+
+      new_cancel_orders = []
+      # check for missing order, or uncancelled old orders
+      if (len(self.buy_for) != self.n_orders):
+        if len(self.buy_for) < self.n_orders:
+          for i in range(len(self.buy_for), self.n_orders):
+            # FIXME not good
+            new_buy_for_prices.append(prices[0][i])
+            new_buy_for_quantities = self._get_quantities(new_buy_for_prices, self.order_size_per_side)
+            # add more orders
+        elif len(self.buy_for) > self.n_orders:
+          extra_buy_for_orders = len(self.buy_for) - self.n_orders
+          while extra_buy_for_orders != 0:
+            new_cancel_orders.append(CancelOrder(market_id=self.market_id, order_hash=self.buy_for.pop().order_hash))
+            extra_buy_for_orders -= 1
+        else:
+          print("Failed to send orders")
+          return
+      if (len(self.buy_against) != self.n_orders):
+        if len(self.buy_against) < self.n_orders:
+          for i in range(len(self.buy_against), self.n_orders):
+            # FIXME: not good
+            new_buy_against_prices.append(prices[1][i])
+            new_buy_against_quantities = self._get_quantities(new_buy_against_prices, self.order_size_per_side)
+            # add more orders
+        elif len(self.buy_against) > self.n_orders:
+          extra_buy_against_orders = len(self.buy_against) - self.n_orders
+          while extra_buy_against_orders != 0:
+            new_cancel_orders.append(
+              CancelOrder(market_id=self.market_id, order_hash=self.buy_against.pop().order_hash)
+            )
+            extra_buy_against_orders -= 1
+        else:
+          print("Failed to send orders")
+          return
+
+      missing_orders = self.get_new_order((new_buy_for_prices, new_buy_against_prices),
+                                          (new_buy_for_quantities, new_buy_against_quantities))
+      if missing_orders:
+        asyncio.run(self.create_orders(missing_orders))
+
+      if new_cancel_orders:
+        asyncio.run(self.cancel_orders(new_cancel_orders))
+
+    # if we exit while loop, we send a cancel all order request.
+    asyncio.run(self.cancel_all_orders())
+
+  def start(self):
+    self.initilization()
+    self.strategy()
+
+
+# FIXME
+# 1 what should I do with 3 states marekt
+# 2 is buy_for price is always lower than buy_against price
+# 3 probabilities data format
+# 4 the insufficient fund case
+# 5 reduce only case
+# 6 when to stop the bot
+# 7 if there is an error, cancel all my current open orders
+
+if __name__ == "__main__":
+  # We assume the probability data is saved in a json file, this file will be updated every n minutes
```

Comment:
we're not really assuming the JSON file gets updated at all. this is just to quickly demonstrate a fake, naive market making algorithm